### PR TITLE
Add support for choreo trajectories

### PR DESCRIPTION
**Pull request by spacey-sooty**
Resolves https://github.com/Mechanical-Advantage/AdvantageScope/issues/269

### CODE REVIEW

**Comment by spacey-sooty** [comment]
It works 
image

File: src/shared/geometry.ts [hunk 1]
```diff
@@ -577,7 +609,7 @@ export function grabHeatmapData(
   logType: string,
   timeRange: "enabled" | "auto" | "teleop" | "teleop-no-endgame" | "full",
   uuid?: string,
-  numberArrayFormat?: "Translation2d" | "Translation3d" | "Pose2d" | "Pose3d",
+  numberArrayFormat?: "Translation2d" | "Translation3d" | "Pose2d" | "Pose3d" | "DifferentialSample" | "SwerveSample",
```
Comment (spacey-sooty) [inline src/shared/geometry.ts:612]: I don't think this makes sense to support for heatmap data but I get type errors if I don't add this

Comment (jwbonner) [reply src/shared/geometry.ts:612]: You don't want to add the types anywhere related to reading number arrays, including `grabNumberArray` (that's only used for the legacy formats). As long as the struct format is supported in `grabPosesAuto` it should work as expected.

File: src/hub/controllers/OdometryController_Config.ts [hunk 2]
```diff
@@ -23,7 +23,11 @@ const OdometryController_Config: SourceListConfig = {
         "Transform2d",
         "Transform3d",
         "Transform2d[]",
-        "Transform3d[]"
+        "Transform3d[]",
+        "DifferentialSample",
```
Comment (jwbonner) [inline src/hub/controllers/OdometryController_Config.ts:27]: I don't think it makes sense to support the non-array version of this as a trajectory type (this applies to the rest of the config too).

File: src/hub/controllers/OdometryController_Config.ts
Comment (jwbonner) [inline src/hub/controllers/OdometryController_Config.ts:1]: Trajectories are also support on the 3D field, so that configuration should be updated too.

**Review by jwbonner — CHANGES_REQUESTED** [review]

File: src/shared/geometry.ts [hunk 3]
```diff
@@ -438,6 +448,21 @@ export function grabPose2d(log: Log, key: string, timestamp: number, uuid?: stri
   ];
 }
 
+export function grabSample(log: Log, key: string, timestamp: number, uuid?: string): AnnotatedPose3d[] {
```
Comment (jwbonner) [inline src/shared/geometry.ts:451]: For clarity, I would rename these `grabChoreoSample` and `grabChoreoSampleArray`. There are a lot of meanings of "sample" in a data visualization tool...

**Review by jwbonner — CHANGES_REQUESTED** [review]

**Comment by spacey-sooty** [comment]
Lit works on 3d field as well

**Review by jwbonner — APPROVED** [review]

**Pull request merged**